src/components/SideNav.js

```diff
@@ -18,34 +18,44 @@ import Logo from "@/assets/repo-rater-logo.svg";
 import GitHub from "@/assets/github-mark.svg";
 import { account } from "@/config/appwrite-client";
 import getUser from "@/utils/github/getUser";
-
-const navigation = [
-  { name: "GitHub Repos", href: "/", icon: FolderIcon, current: false },
-  {
-    name: "Popular Repos",
-    href: "/popular",
-    icon: ServerIcon,
-    current: false,
-  },
-  {
-    name: "User Rankings",
-    href: "/rankings",
-    icon: ChartBarSquareIcon,
-    current: false,
-  },
-  {
-    name: "Star us on GitHub",
-    href: "https://github.com/EddieHubCommunity/RepoRater",
-    icon: StarIcon,
-    current: false,
-    external: true,
-    css: "text-yellow-400",
-  },
-];
+import { usePathname } from "next/navigation";
 
 export default function SideNav({ setKeyword, children }) {
   const [sidebarOpen, setSidebarOpen] = useState(false);
   const [user, setUser] = useState(null);
+  const pathName = usePathname();
+
+  const navigation = useMemo(
+    () => [
+      {
+        name: "GitHub Repos",
+        href: "/",
+        icon: FolderIcon,
+        current: pathName === "/",
+      },
+      {
+        name: "Popular Repos",
+        href: "/popular",
+        icon: ServerIcon,
+        current: pathName === "/popular",
```

Comment:
This seems like a duplicate 🤔 Can the data be kept outside the function as before and have the calculation in the function?

Reply:
Yup. I think we can keep the data outside and do data.map inside the function to fill the current key in the object based on the pathname

Reply:
Done